forge-gui/res/cardsfolder/upcoming/rip_spawn_hunter.txt

```diff
@@ -0,0 +1,8 @@
+Name:Rip, Spawn Hunter
+Types:Legendary Creature Human Survivor
+ManaCost:2 G W
+PT:4/4
+T:Mode$ Phase | Phase$ Main2 | ValidPlayer$ You | PresentDefined$ Self | IsPresent$ Card.tapped | TriggerZones$ Battlefield | Execute$ TrigDig | TriggerDescription$ Survival — At the beginning of your second main phase, if CARDNAME is tapped, reveal the top X cards of your library, where X is its power. Put any number of creature and/or Vehicle cards with different powers from among them into your hand. Put the rest on the bottom of your library in a random order.
+SVar:TrigDig:DB$ Dig | DigNum$ X | AnyNumber$ True | WithDifferentPowers$ True | ChangeValid$ Creature,Vehicle | RestRandomOrder$ True | ForceRevealToController$ True

```

Comment:
WithDifferentPowers$ doesn't exist here

Reply:
Well, it kinda needs to for the card to work... or do we have a workaround?

Reply:
Since it's provided by ChooseCard you could probably emulate it by wrapping that with PeekAndReveal and ChangeZones after

Reply:
Hm... I could have a look tomorrow. Would it be hard to add WithDifferentPowers$ to work here?

Reply:
not really difficult, while I don't have enough time right now maybe another dev can make it happen

Reply:
@Simisays
can you try to script it with ChooseCard?

Reply:
I gave it a try !